File: examples/src/services/order/order.js

const db = require('../../core/db')

exports.findOrder = async ({ id }) => {
  await db.delay(100) // fake db call
  return {
    id,
    items: [{
      id: 1,
      name: 'xbox',
      price: 500
    }]
  }
}

exports.listOrdersForUser = async ({ userId }) => {
  await db.delay(100) // fake db call
  return {
    orders: [
      {
        id: 1,
        items: [{
          id: 1,
          name: 'xbox',
          price: 500
        }]
      }
    ]
  }
}

File: examples/python/libra/services/order/order.py
from libra.core import db

def find_order(id):
    return {
        'id': id,
        'items': [{
            'id': 1,
            'name': 'xbox',
            'price': 500
        }]
    }

def list_orders_for_user(userId):
    return {
        'orders': [{
            'id': 1,
            'items': [{
                'id': 1,
                'name': 'xbox',
                'price': 500
            }]
        }]
    }
}
